x-pack/legacy/plugins/ingest/server/libs/adapters/configurations/adapter.contract.test.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Slapshot from '@mattapperson/slapshot';
+
+import { SODatabaseAdapter } from '../so_database/default';
+import { MockedSODatabaseAdapter } from '../so_database/mocked_adapter';
+import { createKibanaServer } from '../../../../../../../test_utils/jest/contract_tests/servers';
+import { ConfigAdapter } from './default';
+import { Root } from '../../../../../../../../src/core/server/root';
+
+const { callWhenOnline } = Slapshot;
+
+let root: Root;
+let adapter: ConfigAdapter;
+
+describe('Configuration Adapter', () => {
+  beforeAll(async () => {
+    await callWhenOnline(async () => {
+      const { kbnServer } = await createKibanaServer();
+      const savedObjectsAdapter = new SODatabaseAdapter(
+        kbnServer.savedObjects,
+        kbnServer.plugins.elasticsearch
+      );
+      const mockedAdapter = new MockedSODatabaseAdapter(savedObjectsAdapter);
+      adapter = new ConfigAdapter(mockedAdapter as SODatabaseAdapter);
+    });
+
+    if (!adapter) {
+      adapter = new ConfigAdapter(new MockedSODatabaseAdapter() as SODatabaseAdapter);
+    }
+  });
+  afterAll(async () => {
+    await root.shutdown();
```

Comment:
Hmmm... root isn't set anywhere but this doesn't fail?

Reply:
This is a multi-part issue. shutdown is not called, because the global shutdown is not finished and times out. Both of these should fail to have even been reached because the logs show the one test failed due to a bad snapshot... but that part is an issue with snapshot... ill fix that upstream and post here when done